inc/namespace.php

```diff
@@ -63,6 +63,7 @@ function bootstrap( array $config ) {
 	add_filter( 'style_loader_tag', __NAMESPACE__ . '\\output_integrity_for_style', 0, 3 );
 	add_action( 'template_redirect', __NAMESPACE__ . '\\send_normal_csp_header' );
 	add_action( 'template_redirect', __NAMESPACE__ . '\\send_report_only_csp_header' );
+	add_filter( 'rest_pre_serve_request', __NAMESPACE__ . '\\restrict_cors_origin' );
```

Comment:
This should hook in earlier to shortcircuit the request; suggest `rest_pre_dispatch` instead, where we can then return an error response if desired.